ioc/credentials.cpp

```diff
@@ -24,9 +24,26 @@ namespace ioc {
  */
 
 Credentials::Credentials(const server::ClientCredentials& clientCredentials) {
-    // Extract host name part (or whole thing if no colon present)
-    auto pos = clientCredentials.peer.find_first_of(':');
-    host = clientCredentials.peer.substr(0, pos);
+    // There are 3 possibilities of addresses
+    // IPv4 wrapped as IPV6 : e.g.: [::ffff:192.168.1.1]:9999
+    // IPv6: e.g. [2001:0db8:85a3:0000:0000:8a2e:0370:7334]:9999
+    // IPv4 only: e.g.: 192.168.1.1:9999
+    auto pos = clientCredentials.peer.find_first_of("[::ffff:");
+    auto end = clientCredentials.peer.find_first_of(']');
+    //IPv4 wrapped as IPv6
+    if (pos != clientCredentials.peer.npos) {
```

Comment:
Question: What happens, if the address does not include "[::ffff:" ?
Will pos be null ?

Reply:
The result of find_first_of will be npos https://en.cppreference.com/w/cpp/string/basic_string/find_first_of

Reply:
Nice - excuse my missing knowledge in C++
The patch then looks good to me.

Reply:
No problem! Thanks for reviewing!